src/datafile.rs

```diff
@@ -64,18 +64,16 @@ pub struct DataFile {
 
 impl DataFile {
     pub fn create(path: &std::path::Path, is_readonly: bool) -> ErrorResult<DataFile> {
-        let datafile: std::fs::File;
-
-        if is_readonly {
```

Comment:
`if` are expressions in rust, so they return their body.

we can simply just do `let foo = if cond { foo } else { bar };` instead